components/App.js

```diff
@@ -0,0 +1,105 @@
+import NotionEditPage from "./NotionEditPage.js";
+import NotionPage from "./NotionPage.js";
+
+export default function App({ $target }) {
+  // 컨테이너 생성
+  const $notionPageContainer = document.createElement("div");
+  const $notionEditPageContainer = document.createElement("div");
+  $notionPageContainer.className = "notionPageContainer";
+  $notionEditPageContainer.className = "notionEditPageContainer";
+
+  // 페이지 추가 버튼 생성
+  const $addNewRootNotion = document.createElement("button");
+  $addNewRootNotion.className = "addNewRootNotion";
+  $addNewRootNotion.textContent = "+ 페이지 추가";
+
+  // 노션 페이지 컨테이너 토글 버튼 생성
+  const $toggleBtn = document.createElement("button");
+  $toggleBtn.className = "toggle-btn1";
+  $toggleBtn.innerHTML = `<i class="fa fa-solid fa-arrow-right" style="color: #000000;"></i>`;
+
+  $target.appendChild($toggleBtn);
+  $target.appendChild($notionPageContainer);
+  $target.appendChild($notionEditPageContainer);
+  $notionPageContainer.appendChild($addNewRootNotion);
+
+  // 노션 페이지 컨포넌트
+  const notionPage = new NotionPage({
+    $target: $notionPageContainer,
+    initialState: [],
+    onClick: (id) => {
+      const { pathname } = window.location;
+      const [, , notionId] = pathname.split("/");
+      if (id !== notionId) {
+        history.pushState(null, null, `/documents/${id}`);
+        this.route();
+      }
+    },
+    onAdd: (id) => {
+      fetchAddNotion(id);
+    },
+    onDelete: async (id) => {
+      await request(`/documents/${id}`, {
+        method: "DELETE",
+      });
+      history.replaceState(null, null, "/");
+      this.route();
+    },
+    $editorPage: $notionEditPageContainer,
+    $toggleBtn,
+  });
+
+  // 노션 에디터 페이지 컴포넌트
+  const editorPage = new NotionEditPage({
+    $target: $notionEditPageContainer,
+    initialState: "",
+    onRender: () => {
+      notionPage.render();
+    },
+  });
+
+  // 라우팅 처리
+  this.route = () => {
+    const { pathname } = window.location;
+
+    if (pathname === "/") {
+      notionPage.render();
+      editorPage.setState("");
+    } else if (pathname.indexOf("/documents/") === 0) {
+      const [, , notionId] = pathname.split("/");
+      editorPage.setState({ notionId });
+    }
+  };
+
+  this.route();
+
+  initRouter(() => this.route());
+
+  // 페이지 추가 후 서버 저장
+  const fetchAddNotion = async (id = null) => {
+    const createdNotion = await request("/documents", {
+      method: "POST",
+      body: JSON.stringify({
+        title: "제목 없음",
+        parent: id,
+      }),
+    });
+    history.pushState(null, null, `/documents/${createdNotion.id}`);
+    this.route();
+    notionPage.render();
+  };
+
+  // 페이지 추가 이벤트 리스너
+  $addNewRootNotion.addEventListener("click", () => {
+    fetchAddNotion();
+  });
+
+  // 노션 페이지 토글 이벤트 리스너
+  $toggleBtn.addEventListener("click", () => {
+    $notionPageContainer.style.transform = "translateX(0)";
+    $notionEditPageContainer.style.marginLeft = "0";
+    $toggleBtn.style.transform = "translateX(-160%)";
```

Comment:
p3;
현재 코드는 스타일을 인라인으로 주입하는 방식이네요.
인라인으로 직접 속성을 지정하는 것보다는 클래스명으로 조정하는 것이 좀더 라이트한 방법일 듯 합니다.

https://ko.javascript.info/styles-and-classes

```suggestion
    $notionPageContainer.classList.add("toggle-on");
    $notionEditPageContainer.classList.add("toggle-on");
    $toggleBtn.classList.add("toggle-on");
```
```css
.notionPageContainer .toggle-on {
  transform: translateX(0);
}
...
```